applications/GeoMechanicsApplication/custom_utilities/ublas_utilities.cpp

```
@@ -22,4 +22,18 @@ Vector UblasUtilities::CreateVector(const std::initializer_list<double>& rInitia
     std::ranges::copy(rInitializerList, result.begin());
     return result;
 }
+
+Matrix UblasUtilities::CreateMatrix(const std::initializer_list<std::initializer_list<double>>& rInitializerList)
+{
+    if (rInitializerList.size() == 0) return {};
+
+    Matrix result(rInitializerList.size(), rInitializerList.begin()->size(), 0.0);
```

Since there are no checks on the sizes of the supplied rows, it may be safer to determine the maximum row length and use that as the number of columns of the matrix. What do you think? Or should we raise an error when the rows don't have one and the same size?

---

Gennady had indeed a similar comment, I added a check now.